### PR TITLE
Add config to disable default formatter

### LINKED ISSUE #889: provideDocumentFormattingEdits still uses default formatter
I am providing my own custom json formatter with `provideDocumentFormattingEdits`. However, when I use it, the default json formatter is still active and so my formatting gets overridden. Can I set my formatter as the only formatter somehow? 

```ts
provideDocumentFormattingEdits(model: Monaco.editor.ITextModel, options: Monaco.languages.FormattingOptions, token: Monaco.CancellationToken): Monaco.languages.TextEdit[] {
  return [
    {
      range: model.getFullModelRange(),
      text: stringify(JSON.parse(model.getValue())),
    },
  ];
},
```

cc @aeschli from https://github.com/Microsoft/monaco-json

### PR DESCRIPTION
**Pull request by domoritz**
fixes Microsoft/monaco-editor#889

### CODE REVIEW

**Comment by domoritz** [comment]
@aeschli is this what you imagined?

**Comment by aeschli** [comment]
Having it as part of `DiagnosticsOptions` is strange.

Maybe a Capabilities property on the `LanguageServiceDefaults`? Also there needs to be code that listens to the change event and register/unregister the formatter.

**Comment by domoritz** [comment]
Thank you for taking a look at my PR. I updated it to follow your suggestions.

**Comment by aeschli** [comment]
Thanks, improvements are good.
I would still suggest to have a new settings options `Capabilities` and have 'formatter` in there, so we can use that object also for other requests to not register some features.

Also, now `onDidChange` is fired, but isn't it necessary to listen to it and register/deregister the services?

**Comment by domoritz** [comment]
Oh, I misunderstood your comment and didn't read Capabilities as `Capabilities`. I'll work on this. 

I had assumed that `this._onDidChange.fire(this)` would reload the services. I guess I have to keep a reference to the services/providers so that I can deregister them, right?

**Comment by domoritz** [comment]
What are the `disposables` for, btw? The array doesn't seem to be used anywhere.

**Comment by domoritz** [comment]
Done. Let me know if this is what you imagined.

Btw, I noticed that the indentation is inconsistent (space vs tabs). What's the preferred indentation?

**Comment by aeschli** [comment]
looks good!
should be tabs.
yes, there's currently no way to uninstall the language supports, so the disposables are not used.

**Comment by domoritz** [comment]
I can remove `disposables` if you want or just leave it. Other than that, I'm done with the PR.

**Comment by aeschli** [comment]
Thanks, I'll take it from here!

**Comment by domoritz** [comment]
Thank you. Monaco is really awesome and especially the JSON support has been phenomenal for the editor we are building (https://vega.github.io/editor/).

File: src/monaco.d.ts [hunk 1]
```diff
@@ -33,22 +33,63 @@ declare module monaco.languages.json {
...
          */
-        readonly enableSchemaRequest? : boolean;
+        readonly enableSchemaRequest?: boolean;
     }
 
-    export interface Capabilities {
+    export interface ModeConfiguration {
         /**
-         * Disable the default JSON formatter.
+         * Defines wheter the built-in documentFormattingEdit provider is enabled.
```
Comment (domoritz) [inline src/monaco.d.ts]: Typo. Should be "whether".

File: src/monaco.d.ts [hunk 2]
```diff
@@ -33,22 +33,63 @@ declare module monaco.languages.json {
...
          */
-        readonly disableDefaultFormatter?: boolean;
+        readonly documentFormattingEdits?: boolean;
+
+        /**
+         * Defines wheter the built-in documentRangeFormattingEdit provider is enabled.
```
Comment (domoritz) [inline src/monaco.d.ts]: same typo as above (also note the same typo below)

**Comment by aeschli** [comment]
@alexandrudima Here's what we ended up with the APIs to enable/disable certain providers in JSON.
Let me know if that makes sense for you.

**Comment by domoritz** [comment]
Any updates on this @aeschli @alexandrudima?

**Review by alexdima — APPROVED** [review]

**Pull request merged**